pyomo/contrib/mindtpy/global_outer_approximation.py

```diff
@@ -108,4 +108,5 @@ def deactivate_no_good_cuts_when_fixing_bound(self, no_good_cuts):
             if self.config.use_tabu_list:
                 self.integer_list = self.integer_list[:valid_no_good_cuts_num]
         except KeyError as e:
-            self.config.logger.error(str(e) + '\nDeactivating no-good cuts failed.')
+            self.config.logger.error(e, exc_info=True)
+            self.config.logger.error('Deactivating no-good cuts failed.')
```

Comment:
How can this fail? Wouldn't this be a `DeveloperError` (if you need an error here at all) since mindtpy can keep track of what cuts it has added?

Reply:
Thanks for pointing this out. I checked the log of the benchmark tests and found this error came out when `timelimit` was reached. Let me try to fix it before merging this PR. Thanks.

Reply:
The `KeyError` happens when no feasible solution has been found within the `timelimit`. I added one more condition for the `fix_dual_bound` function in the **OA, GOA, LP/NLP B&B and GLP/NLP B&B** methods. 
I also removed `fix_dual_bound` for the **ECP** method since we will never add no-good cuts in the **ECP** method.
I would appreciate it if @bernalde can give it a double-check.